scripts/ajexport.js

```diff
@@ -0,0 +1,106 @@
+  throw new Error('Failed to load Animated Java plugin before CLI plugin');
+}
+const paths = getConfigPaths(
+  'C:\\Users\\Aidan\\Documents\\Media_Storage\\active_projects\\flowey_remaster\\omega-flowey-minecraft-remastered\\scripts\\config.json',
+);
+const dir = paths[3].concat('/');
+console.log('Target paths: ', paths);
+const files = fs.readdirSync(dir).filter((file) => file.includes('ajmodel'));
+const exportNextFile = () => {
+  if (Project) {
+    Project.close();
+  }
+  const file = files.pop();
+  if (typeof file === 'undefined') {
+    return;
+  }
+  consoleLogJson({ file });
+  if (file.includes('ajmodel')) {
+    let content = fs.readFileSync(dir.concat(file), 'utf-8');
+    let name = file.split('/').pop();
+    let fileObj = {
+      path: file,
+      content: injectModelPackPaths(content, paths),
+      name: name,
+    };
+    loadModelFile(fileObj);
+    AnimatedJava.API.safeExportProject(exportNextFile);
+  }
+};
+exportNextFile();
+
+/*
+function getModelPackPaths(modelContent) {
+  var f = JSON.parse(modelContent);
+  var resourcePackPath = f.animated_java.settings.resource_pack_mcmeta;
+  var dataPackPath =
+    f.animated_java.exporter_settings['animated_java:datapack_exporter']
+      .datapack_mcmeta;
+  return [resourcePackPath, dataPackPath];
+}
+*/
+
+/*
+function writeModelPackPaths(modelContent, modelFile, paths) {
+  var f = JSON.parse(modelContent);
+  f.animated_java.settings.resource_pack_mcmeta = paths[0];
+  f.animated_java.exporter_settings[
+    'animated_java:datapack_exporter'
+  ].datapack_mcmeta = paths[1];
+  fs.writeFileSync(modelFile, JSON.stringify(f));
+}
+*/
+
+function injectModelPackPaths(modelContent, paths) {
+  let f = JSON.parse(modelContent);
+  f.animated_java.settings.resource_pack_mcmeta = paths[0];
+  f.animated_java.exporter_settings[
+    'animated_java:datapack_exporter'
+  ].datapack_mcmeta = paths[1];
+  for (const texture of f.textures) {
+    texture.path = texture.path.replaceAll('\\', '/');
+    // TODO this should maybe be regex?
+    if (texture.path.includes('.minecraft')) {
+      const x = texture.path.split('assets')[1];
+      const newPath = `${paths[2]}/assets${x}`;
+      // consoleLogJson({
+      //     texturePath: texture.path,
+      //     afterAssets: x,
+      //     newPath,
+      // });
+      texture.path = newPath;
+    } else if (texture.path.includes('resourcepack/assets')) {
+      const x = texture.path.split('resourcepack/assets')[1];
+      const resourcePackBase = paths[0].split('resourcepack')[0];
+      const newPath = `${resourcePackBase}resourcepack/assets${x}`;
+      // consoleLogJson({
+      //     texturePath: texture.path,
+      //     resourcePackBase,
+      //     afterAssets: x,
+      //     newPath,
+      // });
+      texture.path = newPath;
+    }
+  }
+  return JSON.stringify(f);
+}
+
+function getConfigPaths(configFile) {
+  const f = JSON.parse(fs.readFileSync(configFile).toString());
```

Comment:
ig `toString()` works on file data buffers. ive been using this tho
```suggestion
  const f = JSON.parse(fs.readFileSync(configFile, 'utf8'));
```

Reply:
[6efb816](https://github.com/TheAfroOfDoom/omega-flowey-minecraft-remastered/pull/64/commits/6efb816e56200daf9da8b351c005286820d4eb3d)